src/ansi-c/library/cprover_contracts.c

```diff
@@ -0,0 +1,4116 @@
+/* FUNCTION: __CPROVER_contracts_car_create */
+
+#ifndef __CPROVER_contracts_write_set_t_defined
+#  define __CPROVER_contracts_write_set_t_defined
+
+typedef struct __CPROVER_contracts_car_t
```

Comment:
Surely there is a better way to do this than repeating this code (are we repeating it? I can't quite tell). Maybe put it in a header and #include it in each place? It just seems not great.